samcli/lib/list/resources/resource_mapping_producer.py

```diff
@@ -0,0 +1,80 @@
+"""
+in progress
+"""
+from typing import Optional, Any
+import dataclasses
+import logging
+import yaml
+import click
+
+from botocore.exceptions import ClientError, NoCredentialsError
+from samtranslator.translator.managed_policy_translator import ManagedPolicyLoader
+from samtranslator.translator.arn_generator import NoRegionFound
+
+from samcli.commands.list.exceptions import SamListLocalResourcesNotFoundError, SamListUnknownClientError
+from samcli.lib.list.list_interfaces import Producer
+from samcli.lib.list.resources.resources_def import ResourcesDef
+from samcli.lib.translate.sam_template_validator import SamTemplateValidator
+from samcli.lib.providers.sam_stack_provider import SamLocalStackProvider
+from samcli.lib.translate.translate_utils import _read_sam_file
+from samcli.lib.translate.exceptions import InvalidSamDocumentException
+from samcli.commands.local.cli_common.user_exceptions import InvalidSamTemplateException
+from samcli.commands.exceptions import UserException
+
+LOG = logging.getLogger(__name__)
+
+
+class ResourceMappingProducer(Producer):
```

Comment:
can we add type hints to the functions here? 🙏

Reply:
Resolved